src/main/java/org/jvnet/hudson/test/BuildWatcher.java

```diff
@@ -112,18 +111,18 @@
 
     private static final class RunningBuild {
 
-        private final AnnotatedLargeText<?> log;
+        private final Run<?,?> r;
         private final OutputStream sink;
         private long pos;
 
         RunningBuild(Run<?,?> r) {
-            log = r.getLogText();
+            this.r = r;
             sink = new LogLinePrefixOutputFilter(System.err, "[" + r + "] ");
         }
 
         synchronized void copy() {
             try {
-                pos = log.writeLogTo(pos, sink);
+                pos = r.getLogText().writeLogTo(pos, sink);
```

Comment:
See [this precedent](https://github.com/jenkinsci/jenkins/blob/3f5fe7f6b9b8dab37e82b360574102c6388729e8/core/src/main/java/hudson/cli/ConsoleCommand.java#L75-L79). `getLogText` in general needs to be called repeatedly as more log output arrives.